src/app/genotype-browser/genotype-browser.component.ts

```diff
@@ -82,7 +83,15 @@ export class GenotypeBrowserComponent implements OnInit, OnDestroy {
     public readonly configService: ConfigService,
     private loadingService: FullscreenLoadingService,
     private datasetsService: DatasetsService,
-  ) { }
+    private router: Router
+  ) {
+    this.router.events.pipe(
+      filter(event => event instanceof NavigationStart)
+    ).subscribe(() => {
+      this.queryService.cancelStreamPost();
+      this.loadingService.setLoadingStop();
+    });
```

Comment:
Same as the other router subscription - would be good to unsubscribe from it in OnDestroy.